lib/ts/Layout.cc

```diff
@@ -140,6 +140,21 @@ Layout::check_runroot()
     runroot_map[str.substr(0, pos)] = str.substr(pos + 2);
   }
 
+  // the case for relocate (use ENV to overwrite records.config values)
+  if (runroot_map["prefix"][0] == '.') {
+    setenv("PROXY_CONFIG_CONFIG_DIR", "sysconfdir", true);
+    setenv("PROXY_CONFIG_LOCAL_STATE_DIR", "localstatedir", true);
+    setenv("PROXY_CONFIG_LOG_LOGFILE_DIR", "logdir", true);
+    setenv("PROXY_CONFIG_BIN_PATH", "bindir", true);
+    setenv("PROXY_CONFIG_PLUGIN_PLUGIN_DIR", "libexecdir", true);
```

Comment:
This doesn't seem very cool to me. Only the operator should be setting config environment overrides. Doing it in the midst of library code seems fragile and likely to eventually cause supposing results.

This code is also seems to basically set arbitrary values if `"prefix"` starts with `'.'` (though testing for `!= '/'` would be a more robust way for checking whether it was a relative path). I think this needs a big code comment by way of explanation.

Reply:
We need these ENV override somewhere when we execute traffic server. The best place to put them is during construction of layout. I can move them (the entire function) to runroot.cc and the runroot handler file will take care of it.